bin/test_exercises.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# ###
+# test_exercises.sh, by Tim Austin (neenjaw) tim@neenjaw.com
+# ###
+# Script is to test each exercise against it's test unit individually
+# against the example solution to prove the test works, and the problem
+# is solvable.
+#
+# This works by creating a copy of the exercises, then for each exercise running
+# `mix test` with some options, suppressing the output, then tallying the results
+# with a brief report.
+#
+# An exit code of 0 indicates all tests successful, 1 indicates an error with at
+# least one exercise.
+# ###
+
+# helper subroutine
+function test_or_tests () {
+    local tests="test"
+    if (( $1 != 1 ))
+    then
+      tests+="s"
+    fi
+    printf "%s" "$tests"
+}
+
+# Initialize counts / array
+test_count=0
+pass_count=0
+fail_count=0
+
+failing_exercises=()
+
+# Store the relative root dir to variable
+relative_root=$(pwd)
+
+# make a copy of exercises
+rm -rf tmp-exercises
+cp -a exercises tmp-exercises
+
+# test each exercise
+for exercise in tmp-exercises/*
+do 
+  if [ -d $exercise ]
+  then
+    cd "$exercise"
+
+    exercise_name=$(basename $exercise)
+    test_count=$((test_count+1))
+
+    printf "Testing: $exercise_name"
+
+    # Move the example into the lib file
+    for file in lib/*.ex
+    do 
+      mv "$file" "${file/.ex/.ex.bkp}"
+    done
+
+    mv example.exs lib/example.ex
+
+    # do testing then based on exit code print the result
+    test_results=$(mix test --color --no-elixir-version-check --include pending)
+
+    if [ "$?" -eq 0 ]
+    then
+      printf " -- \\033[32mPass\\033[0m\n"
+      pass_count=$((pass_count+1))
+    else
+      printf " -- \\033[31mFail\\033[0m\n"
+
+
+      test_results=$(echo "$test_results" \
+                      | grep "* test" \
+                      | sed 's:\r:\n:' \
+                      | awk 'NR % 2 == 1' \
+                      | sed -r "s:\x1B\[31m:$(printf " -- \\033[31mFail\\033[0m"):g" \
+                      | sed -r "s:\x1B\[32m:$(printf " -- \\033[32mPass\\033[0m"):g")
+      printf "${test_results}\n"
```

Comment:
I'm not sure this does what we want in a portable way. Running this on mac on elixir 1.8.2/erlang 22 `grep "* test" \` comes up empty because failing tests are printed like this: `1) test`. My version of `sed` also does not have an `-r` option. And at least for the way my `mix test` output if formatted, this would lose the stacktrace in the `grep`.

Reply:
Interesting... 

I suppose then for now can have it just print the whole output from the failing test.  The most robust way I think to have it do something more than that is to write a custom `ExUnit.CLIFormatter` module.